docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-Vasyl-Teliman.md

```diff
@@ -0,0 +1,342 @@
+- [#3622](https://github.com/KhronosGroup/SPIRV-Tools/pull/3622): Fix memory-related bugs in various transformations.
+- [#3640](https://github.com/KhronosGroup/SPIRV-Tools/pull/3640): Handle `OpPhi` instructions during constant obfuscation.
+- [#3642](https://github.com/KhronosGroup/SPIRV-Tools/pull/3642): Handle `OpPhi` instructions in livesafe functions.
+- [#3651](https://github.com/KhronosGroup/SPIRV-Tools/pull/3651): Handle capabilities during module donation.
+- [#3685](https://github.com/KhronosGroup/SPIRV-Tools/pull/3685): Fix bit width in `FuzzerUtilAddEquationInstructions`.
+- [#3689](https://github.com/KhronosGroup/SPIRV-Tools/pull/3689): Support identical predecessors in `TransformationPropagateInstructionUp`.
+- Refactor the fact manager. This is not finished (more PRs will be created):
+  - [#3699](https://github.com/KhronosGroup/SPIRV-Tools/pull/3699): Split the fact manager into multiple files.
+- [#3740](https://github.com/KhronosGroup/SPIRV-Tools/pull/3740): Fix `MaybeGetZeroConstant` function.
+- [#3700](https://github.com/KhronosGroup/SPIRV-Tools/pull/3700): Add support for memory instructions to `TransformationMoveInstructionDown`.
+- [#3729](https://github.com/KhronosGroup/SPIRV-Tools/pull/3729): Skip unreachable blocks in the fuzzer.
+- [#3736](https://github.com/KhronosGroup/SPIRV-Tools/pull/3736): Add support for `BuiltIn` decoration.
+- [#3742](https://github.com/KhronosGroup/SPIRV-Tools/pull/3742): Handle non-existent ids in the fact manager.
+- [#3630](https://github.com/KhronosGroup/SPIRV-Tools/pull/3630): Remove `OpFunctionCall` operands in correct order.
+
+### Potential ideas for the future
+SOm interesting ideas might be worthwhile implementing soon. Those include:
+
+[#3695](https://github.com/KhronosGroup/SPIRV-Tools/issues/3695): Transformation to inline pointer parameters.
+
+[#3723](https://github.com/KhronosGroup/SPIRV-Tools/issues/3723): Implement alias analysis.
+
+[#3696](https://github.com/KhronosGroup/SPIRV-Tools/issues/3696): Introduce context-dependent facts.
+
+[#3725](https://github.com/KhronosGroup/SPIRV-Tools/issues/3725): Determine if the function is pure.
+
+### Challenges
+Some of the implemented transformations were quite challenging. Most of the challenges came from various rules of the SPIR-V language that
+must be satisfied. Specific transformations and various challenging implementation details are mentioned below.
+
+#### [#3477](https://github.com/KhronosGroup/SPIRV-Tools/pull/3477): Permute instructions in the basic block
+There are simple algorithms that produce a random permutation of the sequence of elements. However, they can't be used to permute instructions in the basic block.
+This is because:
+1. Domination rules must be satisfied. That being said, every definition must dominate (read 'precede') all of its users. There are exceptions to this rule but
```

Comment:
"That being said" usually means something like "despite this". I.e. it usually means you are saying something that completely conflicts with the previous sentence. I would replace with "Thus" or something else.

Reply:
Fixed in a597e81. Thanks for pointing out.